### PR TITLE
[INTEGRATION][AIRFLOW] fix issue with using Airflow integration when extra dependencies are missing

### PR DESCRIPTION
**Pull request by lukaszlaszko**
### Problem

Currently, Airflow integration can't be used unless all extra dependencies for registered extractors are installed. This is particularly impractical in situation when neither Google BigQuery nor Great Expectations are used by Airflow deployment. 

Closes: #438

### Solution

As a solution import errors are suppressed when extractors are registered into [openlineage.airflow.extractors._extractors](https://github.com/lukaszlaszko/OpenLineage/blob/airflow/allow-use-without-extras/integration/airflow/openlineage/airflow/extractors/extractors.py#L8) and [openlineage.airflow.extractors._patchers](https://github.com/lukaszlaszko/OpenLineage/blob/airflow/allow-use-without-extras/integration/airflow/openlineage/airflow/extractors/extractors.py#L17). This will prevent startup import failure and eliminate extractors with missing dependencies.

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [x] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)

### CODE REVIEW

File: integration/airflow/openlineage/airflow/extractors/extractors.py [hunk 1]
```diff
@@ -3,22 +3,38 @@
...
 
-_extractors = [
-    PostgresExtractor,
-    BigQueryExtractor,
-    GreatExpectationsExtractor,
-    SnowflakeExtractor
-]
+_extractors = list(
+    filter(
+        None,
```
Comment (mobuchowski) [inline integration/airflow/openlineage/airflow/extractors/extractors.py:10]: Maybe explicit: `lambda x: x is not None`

**Review by mobuchowski — CHANGES_REQUESTED** [review]
Thanks for the contribution @lukaszlaszko!

I'd like to merge your change - after fixing line length to make the linter happy.

File: integration/airflow/openlineage/airflow/extractors/extractors.py (hunk 1 above)
Comment (lukaszlaszko) [reply integration/airflow/openlineage/airflow/extractors/extractors.py]: Done

**Comment by mobuchowski** [comment]
@lukaszlaszko those changes make tests pass:
https://github.com/OpenLineage/OpenLineage/commit/31dff9a4574076fe6aaba9ba4f8565fa4036f10e

**Comment by mobuchowski** [comment]
Closed in https://github.com/OpenLineage/OpenLineage/pull/454

**Pull request closed without merging**